drivers/ethernet/eth_sam_gmac.c

```diff
@@ -49,6 +49,10 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 /*
  * Verify Kconfig configuration
  */
+#if CONFIG_ETH_SAM_GMAC_QUEUES > DT_INST_PROP(0, num_queues)
+#error Number of active hardware TX and RX queues is invalid
+#endif
```

Comment:
I wonder if using `BUILD_ASSERT` would be better.

One for "Number of active hardware TX and RX queues cannot be 0"
Another for "Number of active hardware TX and RX queues is invalid" or "... exceeds the maximum queue count supported in hardware"